lib/model/query/audits.js

```diff
@@ -12,16 +12,25 @@ const { map, mergeLeft } = require('ramda');
 const { Actee, Actor, Audit, Dataset, Entity, Form, Project, Submission } = require('../frames');
 const { extender, sqlEquals, page, QueryOptions, unjoiner } = require('../../util/db');
 const Option = require('../../util/option');
+const Problem = require('../../util/problem');
 const { construct } = require('../../util/util');
 
+const xActionNotes = 'x-action-notes';
 
 const log = (actor, action, actee, details) => ({ run, context }) => {
   const actorId = Option.of(actor).map((x) => x.id).orNull();
   const acteeId = Option.of(actee).map((x) => x.acteeId).orNull();
   const processed = Audit.actionableEvents.includes(action) ? null : sql`clock_timestamp()`;
-  const notes = (context == null) ? null :
-    context.headers['x-action-notes'] == null ? null :
-    decodeURIComponent(context.headers['x-action-notes']); // eslint-disable-line indent
+
+  const notes = (() => {
+    const raw = context?.headers[xActionNotes];
+    if (!raw) return null;
+    try {
+      return decodeURIComponent(raw);
+    } catch (err) {
+      throw Problem.user.invalidHeader({ field: xActionNotes, value: raw });
```

Comment:
Not sure about this, but is there a reason to avoid reflecting the value back to the user? We use `X-Action-Notes` for comments about submissions, so it's possible for the header to contain sensitive information.

Reply:
`invalidHeader` seemed an exact fit for this error, and it requires a `value` field.  Happy to rewrite without the returned `value` if that's preferred.

Reply:
> Maybe it'd actually be helpful to return `value` so that the user can see exactly what they sent.

I think it's reasonable for a client to know what it sent :upside_down_face: